hotrod/src/main/java/io/hyperfoil/hotrod/connection/HotRodConnectionPoolImpl.java

```diff
@@ -0,0 +1,74 @@
+package io.hyperfoil.hotrod.connection;
+
+import java.lang.reflect.Field;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.client.hotrod.impl.transport.netty.ChannelFactory;
+
+import io.hyperfoil.hotrod.api.HotRodConnectionPool;
+import io.netty.channel.EventLoop;
+import io.netty.channel.EventLoopGroup;
+
+public class HotRodConnectionPoolImpl implements HotRodConnectionPool {
+
+   private final HotRodClientPoolImpl hotRodClientPool;
+   private final EventLoop eventLoop;
+   private RemoteCacheManager remoteCacheManager;
+   private RemoteCache remoteCache;
```

Comment:
I can't call `RCM.getCache(...)` because it will block.
I was thinking to allow multiple caches by `public void start(List<String> cacheNames)`, do the `RCM.getCache(...)` and store the result in a Map.
Also, `RemoteCache getRemoteCache()` should be replaced to do a get in the `Map`.
WDYT?

Reply:
`RCM.getCache()` will block only when it's called the first time; afterwards it should just fetch the object from a map. So you need to provide a whitelist list of all caches that could be used, call `getCache()` when initializing the plugin, and then probably check in the pool if the `invoke()` is really accessing one of those white-listed caches. Hmm, such a whitelist is effectively the same as map... so ok, it makes sense to have your own map :)

Reply:
Renamed to `HotRodRemoteCachePool`